src/google-map/google-map.service.ts

```diff
@@ -0,0 +1,56 @@
+import { Injectable } from '@nestjs/common';
+import {
+  Client,
+  GeocodeResponse,
+  PlaceAutocompleteResponse,
+  PlaceAutocompleteResult,
+} from '@googlemaps/google-maps-services-js';
+import { ConfigService } from '@nestjs/config';
+import { get, map } from 'lodash';
+import {
+  AutoCompleteResponse,
+  GetPlaceCoordinates,
+} from './types/AutoComplete';
+
+@Injectable()
+export class GoogleMapService extends Client {
+  private readonly accessKey = this.config.get('GOOGLE_MAPS_ACCESS_KEY');
+
+  constructor(private config: ConfigService) {
+    super();
```

Comment:
Question: I can't remember the syntax but isn't supposed to be an inject decorator here?  🤔  to inject the config 
```suggestion

  constructor(@Inject private config: ConfigService) {
    super();
```

Reply:
this not required for `ConfigService`